driver-sync/src/test/functional/com/mongodb/client/JsonPoweredCrudTestHelper.java

```diff
@@ -557,6 +557,10 @@ BsonDocument getInsertManyResult(final BsonDocument collectionOptions, final Bso
             }
             return toResult(new BsonDocument("insertedIds", insertedIds));
         } catch (MongoBulkWriteException e) {
+            // For transaction tests, the exception is expected to be returned.
+            if (clientSession != null && clientSession.hasActiveTransaction()) {
+                throw e;
```

Comment:
Not sure if this is necessary (here or in async).  I ran the tests in the debugger and never hit this line.

Reply:
The test https://github.com/mongodb/specifications/blob/af26e21a9f7dd4ef316a67a15a1a00675054a533/source/transactions/tests/error-labels.json#L20 requires this check.

Reply:
Ugh, that's awful.  This is a flaw in our shared test infrastructure, but let's not deal with it now.  I opened https://jira.mongodb.org/browse/JAVA-3247 to track it.